contracts/app/andromeda-app-contract/src/reply.rs

```diff
@@ -13,21 +10,17 @@ pub fn on_component_instantiation(deps: DepsMut, msg: Reply) -> Result<Response,
 
     let addr_str = get_reply_address(msg)?;
     let addr = &deps.api.addr_validate(&addr_str)?;
-    ADO_ADDRESSES.save(deps.storage, &descriptor.name, addr)?;
-
-    let mut resp = Response::default();
-
-    if !descriptor.name.starts_with('.') {
-        let kernel_address = ADOContract::default().get_kernel_address(deps.storage)?;
-        let register_component_path_msg = execute::register_component_path(
-            kernel_address,
-            &deps.querier,
-            descriptor.name,
-            addr.clone(),
-        )?;
-
-        resp = resp.add_submessage(register_component_path_msg)
-    }
+    let saved_addr = ADO_ADDRESSES.load(deps.storage, &descriptor.name)?;
+    ensure_eq!(
+        addr,
+        saved_addr,
+        StdError::generic_err(format!(
```

Comment:
Why are we using this error instead of `ContractError`?

Reply:
I'll make a new ContractError for this case